feat: GutenbergKit Media Library support

dcalhoun opened the pull request:
## Related

* https://github.com/Automattic/dotcom-forge/issues/9086
* https://github.com/wordpress-mobile/GutenbergKit/pull/36

## Description

Add support for attaching Media Library items to media blocks—Image, Gallery, Video, etc.

## Testing Instructions

See https://github.com/wordpress-mobile/GutenbergKit/pull/36. 

<!-- Test instructions per dependency update: https://github.com/wordpress-mobile/WordPress-Android/blob/trunk/docs/test_instructions_per_dependency_update.md -->

## Regression Notes

1. Potential unintended areas of impact
    Regressions in Gutenberg Mobile or Aztec media attachments. 
2. What I did to test those areas of impact (or what existing automated tests I relied on)
    Manually tested attaching media. 
3. What automated tests I added (or what prevented me from doing so)
    Deemed unnecessary for the experimental GutenbergKit editor. 

## PR Submission Checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

## Testing Checklist (strike-out the not-applying and unnecessary ones):

- [ ] WordPress.com sites and self-hosted Jetpack sites.
- [ ] Portrait and landscape orientations.
- [ ] Light and dark modes.
- [ ] Fonts: Larger, smaller and bold text.
- [ ] High contrast.
- [ ] Talkback.
- [ ] Languages with large words or with letters/accents not frequently used in English.
- [ ] Right-to-left languages. (Even if translation isn’t complete, formatting should still respect the right-to-left layout)
- [ ] Large and small screen sizes. (Tablet and smaller phones)
- [ ] Multi-tasking: Split screen and Pop-up view. (Android 10 or higher)

dcalhoun committed:
wip: GutenbergKit media library support

dcalhoun committed:
feat: Scope Media Library selection by media type

Support Gutenberg's `allowedTypes` parameter so that invalid media is
not displayed in the media picker.

dcalhoun committed:
feat: Attach media to GutenbergKit blocks

dangermattic commented:
<!--
  0 Errors
  1 Warning: PR is not assigned to a milest...
  0 Messages
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">PR is not assigned to a milestone.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

dcalhoun committed:
build: Update GutenbergKit ref

dcalhoun committed:
build: Update GutenbergKit ref

dcalhoun committed:
docs: Add reminder comment

dcalhoun committed:
feat: Support initial Media Library selection

Align UX with the web Media Library and support the web Gallery block
implementation, where previously selected media must be sent back to the
block.

dcalhoun committed:
build: Update GutenbergKit ref

dcalhoun committed:
docs: Remove outdated comment

dcalhoun committed:
style: Remove unused import

dcalhoun committed:
style: Shorten line length

Address lint error.

dcalhoun committed:
fix: Disable non-network image appending

GutenbergKit relies upon the WebView and Gutenberg web logic for file
uploads, so upload tracking is irrelevant. Additionally, until we build
support for sharing files to the WordPress/Jetpack app, we early return.

dcalhoun committed:
refactor: Remove unnecessary safe call on non-null type

This value became non-null in a separate change.

dcalhoun committed:
refactor: Share common logic, remove unused logic

dcalhoun committed:
refactor: Fix typo

dcalhoun committed:
fix: Correctly access single initial selection value

Casting `Value` to `Int` is not possible, must reference the value
itself.

dcalhoun committed:
build: Update GutenbergKit ref

nbradbury commented:
Feedback provided in GutenbergKit repo [here](https://github.com/wordpress-mobile/GutenbergKit/pull/36#issuecomment-2476271955).

nbradbury reviewed (approved):
@dcalhoun I'll approve this and you can merge whenever you're ready :shipit:

dcalhoun committed:
build: Update GutenbergKit ref
